src/array.cr

```diff
@@ -862,11 +862,30 @@ class Array(T)
     self
   end
 
+  # Select array elements that meets conditions from `self`. Returns `self`.
+  # 
+  # if no elements found, self become an empty array.
+  # ```
+  # ary = [1,2,3,4]
+  # ary.select!{|elem| elem % 2 == 0 }  #=>[2,4]
+  # ary                                 #=>[2,4]
+  # ary.select!{|elem| elem % 2 == 1 }  #=>[]
+  # ary                                 #=>[]
+  # ```
   def select!
     delete_if { |elem| !(yield elem) }
     self
   end
 
+  # Select array elements that doesn't meet conditions from `self`. Returns `self`.
```

Comment:
How about "Removes all elements from `self` for which the given block returns a falsy value"?